odoo/addons/base/tests/common.py

```diff
@@ -354,19 +354,25 @@ def _send_email(self, msg, smtp_session):
             self.env['ir.mail_server'].send_email(msg, smtp_session=smtp_session)
         return smtp_session.messages.pop()
 
-    def assertSMTPEmailsSent(self, smtp_from=None, smtp_to_list=None, message_from=None, from_filter=None, emails_count=1):
-        """Check that the given email has been sent.
-
-        If one of the parameter is None, it's just ignored and not used to retrieve the email.
+    def assertSMTPEmailsSent(self, smtp_from=None, smtp_to_list=None, message_from=None,
+                             mail_server=None, from_filter=None,
+                             emails_count=1):
+        """Check that the given email has been sent. If one of the parameter is
+        None it is just ignored and not used to retrieve the email.
 
         :param smtp_from: FROM used for the authentication to the mail server
         :param smtp_to_list: List of destination email address
         :param message_from: FROM used in the SMTP headers
+        :arap mail_server: used to compare the 'from_filter' as an alternative
+          to using the from_filter parameter
         :param from_filter: from_filter of the <ir.mail_server> used to send the email
-            Can use a lambda to check the value
         :param emails_count: the number of emails which should match the condition
         :return: True if at least one email has been found with those parameters
         """
+        if from_filter and mail_server:
```

Comment:
False is a valid value from from_filter (means "match everything")
```suggestion
        if from_filter is not None and mail_server: